modules/openapi-generator/src/main/resources/dart-dio/class.mustache

```diff
@@ -22,5 +22,12 @@ abstract class {{classname}} implements Built<{{classname}}, {{classname}}Builde
 
     factory {{classname}}([updates({{classname}}Builder b)]) = _${{classname}};
     static Serializer<{{classname}}> get serializer => _${{classVarName}}Serializer;
-
 }
+{{! Generate an enum for any variables that are declared as inline enums}}
```

Comment:
why this is only for inline enums? look like the is no check for inline so all enum will be handle like this no ?

Reply:
I think if the enum is actually a schema, then isEnum returns false. I'll need to check that though.

Reply:
@jaumard sorry the delay, finally had time to look into this to confirm.

isEnum is false for enums that are declared as definitions rather than inline, so only inline enums will be affected.

Here's the output from generating petStore with debugModels true:

<details>
  <summary>PetStatus as inline enum</summary>
```
{
  "openApiType" : "string",
  "baseName" : "status",
  "getter" : "getStatus",
  "setter" : "setStatus",
  "description" : "pet status in the store",
  "dataType" : "String",
  "datatypeWithEnum" : "StatusEnum",
  "name" : "status",
  "defaultValue" : "null",
  "defaultValueWithParam" : " = data.status;",
  "baseType" : "String",
  "unescapedDescription" : "pet status in the store",
  "example" : "null",
  "jsonSchema" : "{\n  \"type\" : \"string\",\n  \"description\" : \"pet status in the store\",\n  \"enum\" : [ \"available\", \"pending\", \"sold\" ]\n}",
  "exclusiveMinimum" : false,
  "exclusiveMaximum" : false,
  "hasMore" : false,
  "required" : false,
  "deprecated" : false,
  "secondaryParam" : false,
  "hasMoreNonReadOnly" : false,
  "isPrimitiveType" : true,
  "isModel" : false,
  "isContainer" : false,
  "isString" : true,
  "isNumeric" : false,
  "isInteger" : false,
  "isLong" : false,
  "isNumber" : false,
  "isFloat" : false,
  "isDouble" : false,
  "isByteArray" : false,
  "isBinary" : false,
  "isFile" : false,
  "isBoolean" : false,
  "isDate" : false,
  "isDateTime" : false,
  "isUuid" : false,
  "isUri" : false,
  "isEmail" : false,
  "isFreeFormObject" : false,
  "isListContainer" : false,
  "isMapContainer" : false,
  "isEnum" : true,
  "isReadOnly" : false,
  "isWriteOnly" : false,
  "isNullable" : true,
  "isSelfReference" : false,
  "isCircularReference" : false,
  "_enum" : [ "available", "pending", "sold" ],
  "allowableValues" : {
    "enumVars" : [ {
      "name" : "available",
      "isString" : true,
      "value" : "\"available\""
    }, {
      "name" : "pending",
      "isString" : true,
      "value" : "\"pending\""
    }, {
      "name" : "sold",
      "isString" : true,
      "value" : "\"sold\""
    } ],
    "values" : [ "available", "pending", "sold" ]
  },
  "vendorExtensions" : {
    "x-index" : 6
  },
  "hasValidation" : false,
  "isInherited" : false,
  "nameInCamelCase" : "Status",
  "nameInSnakeCase" : "STATUS",
  "enumName" : "StatusEnum",
  "uniqueItems" : false,
  "isXmlAttribute" : false,
  "isXmlWrapped" : false,
  "datatype" : "String",
  "iexclusiveMaximum" : false
} ]
```
</details>

<details>
  <summary>PetStatus as definition</summary>
```
{
      "openApiType" : "PetStatus",
      "baseName" : "status",
      "complexType" : "PetStatus",
      "getter" : "getStatus",
      "setter" : "setStatus",
      "dataType" : "PetStatus",
      "datatypeWithEnum" : "PetStatus",
      "name" : "status",
      "defaultValue" : "null",
      "defaultValueWithParam" : " = data.status;",
      "baseType" : "PetStatus",
      "example" : "null",
      "jsonSchema" : "{\n  \"$ref\" : \"#/components/schemas/PetStatus\"\n}",
      "exclusiveMinimum" : false,
      "exclusiveMaximum" : false,
      "hasMore" : false,
      "required" : false,
      "deprecated" : false,
      "secondaryParam" : false,
      "hasMoreNonReadOnly" : false,
      "isPrimitiveType" : false,
      "isModel" : false,
      "isContainer" : false,
      "isString" : false,
      "isNumeric" : false,
      "isInteger" : false,
      "isLong" : false,
      "isNumber" : false,
      "isFloat" : false,
      "isDouble" : false,
      "isByteArray" : false,
      "isBinary" : false,
      "isFile" : false,
      "isBoolean" : false,
      "isDate" : false,
      "isDateTime" : false,
      "isUuid" : false,
      "isUri" : false,
      "isEmail" : false,
      "isFreeFormObject" : false,
      "isListContainer" : false,
      "isMapContainer" : false,
      "isEnum" : false,
      "isReadOnly" : false,
      "isWriteOnly" : false,
      "isNullable" : true,
      "isSelfReference" : false,
      "isCircularReference" : false,
      "allowableValues" : {
        "enumVars" : [ {
          "name" : "available",
          "isString" : true,
          "value" : "\"available\""
        }, {
          "name" : "pending",
          "isString" : true,
          "value" : "\"pending\""
        }, {
          "name" : "sold",
          "isString" : true,
          "value" : "\"sold\""
        } ],
        "values" : [ "available", "pending", "sold" ]
      },
      "vendorExtensions" : {
        "x-index" : 6
      },
      "hasValidation" : false,
      "isInherited" : false,
      "nameInCamelCase" : "Status",
      "nameInSnakeCase" : "STATUS",
      "uniqueItems" : false,
      "isXmlAttribute" : false,
      "isXmlWrapped" : false,
      "datatype" : "PetStatus",
      "iexclusiveMaximum" : false
    } ],
```
</details>